editline: 1.16.1 -> 1.17.0

r-ryantm opened the pull request:
Semi-automatic update generated by https://github.com/ryantm/nixpkgs-update tools. This update was made based on information from https://repology.org/metapackage/editline/versions.



meta.description for editline is: '"A readline() replacement for UNIX without termcap (ncurses)"'.


meta.homepage for editline is: '"http://troglobit.com/editline.html"

[Release on GitHub](https://github.com/troglobit/editline/releases/tag/1.17.0)


[Compare changes on GitHub](https://github.com/troglobit/editline/compare/1.16.1...1.17.0)

<details>
<summary>
Checks done (click to expand)
</summary>

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed

- 0 of 0 passed binary check by having a zero exit code.
- 0 of 0 passed binary check by having the new version present in output.
- found 1.17.0 with grep in /nix/store/0mfakwbbdffgn45qia2ykr0xy86n1m63-editline-1.17.0
- directory tree listing: https://gist.github.com/d521163e56489c5687bdd86e24722264

- du listing: https://gist.github.com/4ca7530a34479270fba910def60eb032


</details>
<details>
<summary>
Rebuild report (if merged into master) (click to expand)
</summary>

208 total rebuild path(s)

58 package rebuild(s)

58 x86_64-linux rebuild(s)
54 i686-linux rebuild(s)
44 x86_64-darwin rebuild(s)
52 aarch64-linux rebuild(s)


First fifty rebuilds by attrpath
bundix
cabal2nix
cachix
common-updater-scripts
crystal2nix
dep2nix
discover
disnix
disnixos
editline
fusionInventory
gnome3.gnome-packagekit
gnome3.gnome-software
gnome3.simple-scan
go2nix
haskellPackages.cachix
haskellPackages.nix-paths
hydra
jush
lispPackages.quicklisp-to-nix
lispPackages.quicklisp-to-nix-system-info
lorri
nix
nix-bundle
nix-du
nix-exec
nix-index
nix-pin
nix-plugins
nix-prefetch
nix-prefetch-bzr
nix-prefetch-cvs
nix-prefetch-docker
nix-prefetch-git
nix-prefetch-hg
nix-prefetch-scripts
nix-prefetch-svn
nix-serve
nix-update-source
nixFlakes
nixStable
nixUnstable
nixos-generators
nixpkgs-review
nixui
packagekit
packagekit-qt
pantheon.appcenter
pantheon.nixpkgs-github-update
plasma5.discover

</details>

<details>
<summary>
Instructions to test this update (click to expand)
</summary>

Either download from Cachix:
```
nix-store -r /nix/store/0mfakwbbdffgn45qia2ykr0xy86n1m63-editline-1.17.0 \
  --option binary-caches 'https://cache.nixos.org/ https://r-ryantm.cachix.org/' \
  --option trusted-public-keys '
  r-ryantm.cachix.org-1:gkUbLkouDAyvBdpBX0JOdIiD2/DP1ldF3Z3Y6Gqcc4c=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(r-ryantm's Cachix cache is only trusted for this store-path realization.)

Or, build yourself:
```
nix-build -A editline https://github.com/r-ryantm/nixpkgs/archive/525e296c2cb19ac2ee66b60bc13ef7c6b5dae003.tar.gz
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/0mfakwbbdffgn45qia2ykr0xy86n1m63-editline-1.17.0
ls -la /nix/store/0mfakwbbdffgn45qia2ykr0xy86n1m63-editline-1.17.0/bin
```


</details>
<br/>




cc @dtzWill for testing.

r-ryantm committed:
editline: 1.16.1 -> 1.17.0

dtzWill commented:
FWIW Nix uses `editline`, probably most important in-tree user (only other is `jush`, by same maintainer/author as `editline` so probably okay if looks okay :wink:).

If basic multi-line support is enabled by default, may be good to put through paces by a few folks to ensure `nix repl` is a good experience.  cc @edolstra as unsure who else to ping on that :).

nh2 reviewed (approved):
`nix-review` output:

<detail>

```
$ git -c fetch.prune=false fetch --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/77050/head:refs/nixpkgs-review/1
remote: Enumerating objects: 5, done.
remote: Counting objects: 100% (5/5), done.
remote: Total 7 (delta 5), reused 5 (delta 5), pack-reused 2
Unpacking objects: 100% (7/7), done.
From https://github.com/NixOS/nixpkgs
 + 70ebe5b46a7...525e296c2cb refs/pull/77050/head -> refs/nixpkgs-review/1  (forced update)
$ git worktree add /home/niklas/.cache/nixpkgs-review/pr-77050/nixpkgs 353ea7f3ebd34a989c30cfa714644b86f1fd50ef
Preparing worktree (detached HEAD 353ea7f3ebd)
Updating files: 100% (20738/20738), done.
HEAD is now at 353ea7f3ebd Merge pull request #77105 from r-ryantm/auto-update/kdbg
$ git merge --no-commit 525e296c2cb19ac2ee66b60bc13ef7c6b5dae003
Automatic merge went well; stopped before committing as requested
$ nix build --no-link --keep-going --option build-use-sandbox relaxed -f /home/niklas/.cache/nixpkgs-review/pr-77050/build.nix
warning: SQLite database '/nix/var/nix/db/db.sqlite' is busy
warning: SQLite database '/nix/var/nix/db/db.sqlite' is busy
warning: SQLite database '/nix/var/nix/db/db.sqlite' is busy
warning: SQLite database '/nix/var/nix/db/db.sqlite' is busy
warning: SQLite database '/nix/var/nix/db/db.sqlite' is busy
error: build of '/nix/store/njz7419pnvn5r8w42shwbv1pgis2ygrz-python3.7-ZODB-5.5.1.drv' on 'ssh://static-haskell-nix-ci' failed: builder for '/nix/store/njz7419pnvn5r8w42shwbv1pgis2ygrz-python3.7-ZODB-5.5.1.drv' failed with exit code 1
builder for '/nix/store/njz7419pnvn5r8w42shwbv1pgis2ygrz-python3.7-ZODB-5.5.1.drv' failed with exit code 1; last 10 log lines:
    Tear down ZODB.tests.util.BlobFileStorageBlobTests in 0.000 seconds.
    Set up ZODB.tests.util.MVCCMappingBlobTests in 0.000 seconds.
    Ran 15 tests with 0 failures, 1 errors and 0 skipped in 0.251 seconds.
  Tearing down left over layers:
    Tear down ZODB.tests.util.MVCCMappingBlobTests in 0.000 seconds.
  
  Test-modules with import problems:
    ZODB.tests.test_datamanageradapter
  Total: 1548 tests, 0 failures, 2 errors and 0 skipped in 6 minutes 46.598 seconds.
  builder for '/nix/store/njz7419pnvn5r8w42shwbv1pgis2ygrz-python3.7-ZODB-5.5.1.drv' failed with exit code 1
error: build of '/nix/store/42f7708bqwy9zmmkryw2ww6a0ywi79gs-nix-2.3pre6895_84de821.drv' failed: builder for '/nix/store/42f7708bqwy9zmmkryw2ww6a0ywi79gs-nix-2.3pre6895_84de821.drv' failed with exit code 2
builder for '/nix/store/42f7708bqwy9zmmkryw2ww6a0ywi79gs-nix-2.3pre6895_84de821.drv' failed with exit code 1; last 10 log lines:
  running test tests/pure-eval.sh... [PASS]
  running test tests/check.sh... [PASS]
  running test tests/plugins.sh... [PASS]
  running test tests/search.sh... [PASS]
  running test tests/nix-copy-ssh.sh... [PASS]
  running test tests/post-hook.sh... [PASS]
  running test tests/function-trace.sh... [PASS]
  1 out of 56 tests failed 
  make: *** [mk/tests.mk:12: installcheck] Error 1
  builder for '/nix/store/42f7708bqwy9zmmkryw2ww6a0ywi79gs-nix-2.3pre6895_84de821.drv' failed with exit code 2
cannot build derivation '/nix/store/3h4w746msifar7lcv6k2q4xaviccd733-disnix-0.9.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/ym9hf60shcbpvk96v6ywkyihb0p82sja-nix-plugins-6.0.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/sp4c9wh5j1bn8ihaxymnhk4zpr25zj6j-disnixos-0.8.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/0m4yc6a6xz0bgniilnq38am7zcj275q2-vulnix-1.9.4.drv': 1 dependencies couldn't be built
waiting for locks or build slots...
error: build of '/nix/store/sa50irgn1d3bx471ng4s5jwwly32fnjp-python3.8-nixpkgs-0.2.4.drv' failed: builder for '/nix/store/sa50irgn1d3bx471ng4s5jwwly32fnjp-python3.8-nixpkgs-0.2.4.drv' failed with exit code 1
builder for '/nix/store/sa50irgn1d3bx471ng4s5jwwly32fnjp-python3.8-nixpkgs-0.2.4.drv' failed with exit code 1; last 10 log lines:
  ModuleNotFoundError: No module named 'nix'
  
  
  ----------------------------------------------------------------------
  Ran 1 test in 0.000s
  
  FAILED (errors=1)
  Test failed: <unittest.runner.TextTestResult run=1 errors=1 failures=0>
  error: Test failed: <unittest.runner.TextTestResult run=1 errors=1 failures=0>
  builder for '/nix/store/sa50irgn1d3bx471ng4s5jwwly32fnjp-python3.8-nixpkgs-0.2.4.drv' failed with exit code 1
waiting for locks or build slots...
cannot build derivation '/nix/store/32yn3727bdpw3fnk6vqqg2dkdvxngs35-env.drv': 6 dependencies couldn't be built
[53 built (3 failed), 1007 copied (7289.9 MiB), 756.6 MiB DL]
error: build of '/nix/store/32yn3727bdpw3fnk6vqqg2dkdvxngs35-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/77050
1 package marked as broken and skipped:
nix-exec

6 package failed to build:
disnix disnixos nix-plugins nixUnstable python38Packages.nixpkgs vulnix

48 package built:
bundix cabal2nix cachix common-updater-scripts crystal2nix dep2nix discover editline fusionInventory gnome3.gnome-packagekit gnome3.gnome-software simple-scan go2nix haskellPackages.cachix haskellPackages.nix-paths hydra jush lispPackages.quicklisp-to-nix lispPackages.quicklisp-to-nix-system-info lorri nix nix-bundle nix-du nix-index nix-pin nix-prefetch nix-prefetch-bzr nix-prefetch-cvs nix-prefetch-docker nix-prefetch-git nix-prefetch-hg nix-prefetch-scripts nix-prefetch-svn nix-serve nix-update-source nixFlakes nixos-generators nixpkgs-review nixui packagekit packagekit-qt pantheon.appcenter pantheon.nixpkgs-github-update python37Packages.nixpkgs python37Packages.pythonix python38Packages.pythonix system-config-printer vgo2nix
```

</detail>

`nix repl` works as expected.

I don't know how to test these multi-line changes, or if there's anything to test about them.

dtzWill commented:
Multi-line changes seem to include at least the following:

* run nix repl
* input something too long for your line, so it wraps to next (adjusting terminal size makes this easier ;))
* try jumping to start/end using control+A or control+E

The cursor jumps across lines now, instead of kinda redrawing in-place somewhat messily (try it with a version of nix before this update to compare :)).

I think we'll want to grab this patch while pulling this in: 2d2f3dabac63f30b8bd1862f5f88cb4efffde757 .

Not adding to PR myself since not entirely confident I understand what this fixes.
Maybe it can wat-- I forget that editline is "cheap" to patch here, compared to libedit which rebuilds all the things :).

dtzWill committed:
editline: patch for improved handling of multiline input

Minor but observable in `nix repl` when history contains
expression that wraps and one that doesn't and using up/down
arrows to toggle between (for example).
